src/components/Profile/EditProfile.js

```diff
@@ -0,0 +1,246 @@
+import React, { useCallback, useMemo, useRef, useState } from 'react'
+import {
+  Button,
+  EthIdenticon,
+  GU,
+  IconCross,
+  Split,
+  shortenAddress,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@1hive/1hive-ui'
+
+import ProfileForm from './ProfileForm'
+import ProposalSupporting from './ProposalSupporting'
+import StakeManagment from './StakesManagment'
+import Tabs from './Tabs'
+import { useProfile } from '../../providers/Profile'
+import { UPLOAD_IPFS_ENDPOINT } from '../../endpoints'
+
+const IMAGE_DIMENSION = 16 * GU
+const CONTENT = [ProfileForm, StakeManagment, ProposalSupporting]
+
+function EditProfile({ onBack }) {
+  const [selectedTab, setSelectedTab] = useState(0)
+  const [profilePic, setProfilePic] = useState({
+    buffer: null,
+    updated: false,
+    removed: false,
+  })
+
+  const theme = useTheme()
+  const { name: layout } = useLayout()
+  const { account, image, name } = useProfile()
+  const oneColumn = layout === 'small' || layout === 'medium'
+
+  const imageInput = useRef(null)
+
+  const handleImageChange = useCallback(event => {
+    const newPhotoFile = event.target.files[0]
+
+    if (newPhotoFile.size >= 2500000) {
+      console.error('File exceeds 2.5 MB')
+    }
+
+    const formData = new FormData()
+    formData.append('path', newPhotoFile)
+
+    setProfilePic({ buffer: formData, updated: true, removed: false })
+  }, [])
+
+  const handleImageRemoval = useCallback(() => {
+    setProfilePic({ buffer: null, updated: false, removed: true })
+  }, [])
+
+  const handlePicFetch = useCallback(async buffer => {
+    const res = await fetch(UPLOAD_IPFS_ENDPOINT, {
```

Comment:
Yeah we probably should.

Note we moved this function into its own file of `services` in #72 since it didn't make much sense to have it inside the component.